opnode/withdrawals/utils.go

```diff
@@ -0,0 +1,266 @@
+package withdrawals
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math/big"
+	"time"
+
+	"github.com/ethereum-optimism/optimistic-specs/l2os/bindings/l2oo"
+	"github.com/ethereum-optimism/optimistic-specs/opnode/contracts/deposit"
+	"github.com/ethereum-optimism/optimistic-specs/opnode/contracts/withdrawer"
+	"github.com/ethereum-optimism/optimistic-specs/opnode/predeploy"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/ethclient/gethclient"
+	"github.com/ethereum/go-ethereum/rlp"
+	"github.com/ethereum/go-ethereum/rpc"
+)
+
+// WaitForFinalizationPeriod waits until the timestamp has been submitted to the L2 Output Oracle on L1 and
+// then waits for the finalization period to be up.
+// This functions polls and can block for a very long time if used on mainnet.
+// This returns the timestamp to use for the proof generation.
+func WaitForFinalizationPeriod(ctx context.Context, client *ethclient.Client, portalAddr common.Address, timestamp uint64) (uint64, error) {
+	opts := &bind.CallOpts{Context: ctx}
+	timestampBig := new(big.Int).SetUint64(timestamp)
+
+	portal, err := deposit.NewOptimismPortalCaller(portalAddr, client)
```

Comment:
Same applies to `deposit` as to `withdrawer` rename I think? Renames in a later PR are fine